src/Menu/Menu.js

```diff
@@ -396,10 +400,27 @@ class Menu extends Component {
         event.preventDefault();
         this.decrementKeyboardFocusIndex();
         break;
+      default:
+        const hotKeys = this.hotKeyHolder.append(key);
+        if (this.setFocusIndexStartsWith(hotKeys)) {
+          event.preventDefault();
+        }
     }
     this.props.onKeyDown(event);
   };
 
+  setFocusIndexStartsWith(keys) {
+    const found = React.Children.toArray(this.props.children).findIndex((child) => {
```

Comment:
I wouldn't using `React.Children.forEach` be more efficient than using `React.Children.toArray.findIndex`?
That's going to be less explicit but we potentianlly have a lot of children. Performance matters here.